src/api/app/controllers/status/checks_controller.rb

```diff
@@ -0,0 +1,75 @@
+class Status::ChecksController < ApplicationController
+  before_action :require_checkable, only: [:index, :show, :destroy, :update]
+  before_action :require_or_initialize_checkable, only: :create
+  before_action :require_check, only: [:show, :destroy, :update]
+  before_action :set_xml_hash, only: [:create, :update]
+  skip_before_action :require_login, only: [:show, :index]
+  after_action :verify_authorized
+
+  def index
+    @checks = @checkable.checks
+    authorize @checks
+  end
+
+  def show
+    authorize @check
+  end
+
+  def create
+    @xml_hash[:checkable] = @checkable
+    @check = Status::Check.new(@xml_hash)
+    authorize @check
+    if @check.save
+      render :show
+    else
+      render_error(status: 422, errorcode: 'invalid_check', message: "Could not save check: #{@check.errors.full_messages.to_sentence}")
+    end
+  end
+
+  def update
+    authorize @check
+    if @check.update(@xml_hash)
+      render :show
+    else
+      render_error(status: 422, errorcode: 'invalid_check', message: "Could not save check: #{@check.errors.full_messages.to_sentence}")
+    end
+  end
+
+  def destroy
+    authorize @check
+    if @check.destroy
+      render_ok
+    else
+      render_error(status: 422, errorcode: 'invalid_check', message: "Could not delete check: #{@check.errors.full_messages.to_sentence}")
+    end
+  end
+
+  private
+
+  def require_or_initialize_checkable
+    project = Project.get_by_name(params[:project_name])
+    repository = project.repositories.find_by!(name: params[:repository_name])
+    @checkable = repository.status_publishes.find_or_initialize_by(build_id: params[:status_repository_publish_build_id])
+  end
+
+  def require_checkable
+    @checkable = Status::RepositoryPublish.find_by(build_id: params[:status_repository_publish_build_id]) if params[:status_repository_publish_build_id]
+    render_error(status: 404, errorcode: 'not_found', message: "Unable to find check with id '#{params[:id]}'") unless @checkable
```

Comment:
> Unable to find check with id

This is not the check, this is the checkable. In the route we call it `status_repository_publish`